v1/proto/openconfig/gsii/v1/interfaces/interfaces.proto

```diff
@@ -0,0 +1,27 @@
+// openconfig.gsii.v1.interfaces is generated by proto_generator as a protobuf
+// representation of a YANG schema.
+//
+// Input schema modules:
+//  - /Users/robjs/go/src/github.com/openconfig/gsii/v1/yang/interfaces/gsii-interfaces-proto.yang
+// Include paths:
+//   - /Users/robjs/go/src/github.com/openconfig/gsii/deps/public/...
+syntax = "proto3";
+
+package openconfig.gsii.v1.interfaces;
+
+import "github.com/openconfig/ygot/proto/ywrapper/ywrapper.proto";
+import "github.com/openconfig/ygot/proto/yext/yext.proto";
+
+option go_package = "github.com/openconfig/gsii/v1/proto/openconfig/gsii/v1/interfaces";
+
+message Interface {
+  ywrapper.BoolValue enabled = 75491684 [(yext.schemapath) = "/interfaces/interface/volatile/enabled"];
+}
+
+message Interfaces {
+  message InterfaceKey {
+    string name = 1 [(yext.schemapath) = "/interfaces/interface/config/name|/interfaces/interface/name"];
+    Interface interface = 2;
```

Comment:
It is intended. YANG lists are generated as `repeated <message>` fields. Where `<message>` is a generated message of the form:

```
message FooKey {
  string key_one = 1;
  string key_two = 2;
  ...
  string key_fourty = 40;
  Foo foo = 41;
}
```

Where `Foo` is the actual "member" of the list, and each of the monotonically increasing fields are the keys in the YANG `list`'s `key` statement. Since these are stably ordered by the YANG schema itself we can save ourselves a few bytes and just give them a number.

For all other fields within a `list` or `container`, we don't have a stable numbering -- so we calculate the field ID using a fnv hash. We published something as to why this was reasonable [here](https://www.tdcommons.org/cgi/viewcontent.cgi?article=1944&context=dpubs_series).

Reply:
cool, sounds good!
thanks for linking that doc